src/hw-monitor.cpp

```diff
@@ -226,10 +225,39 @@ namespace librealsense
     }
 
 
-    void hw_monitor::get_gvd(size_t sz, unsigned char* gvd, uint8_t gvd_cmd) const
+    void hw_monitor::get_gvd( size_t sz,
+                              unsigned char * gvd,
+                              uint8_t gvd_cmd,
+                              const std::set< int32_t > * retry_error_codes ) const
     {
         command command(gvd_cmd);
-        auto data = send(command);
+        hwmon_response p_response = hwmon_response::hwm_Unknown;
+        auto data = send( command, &p_response );
+        if( p_response != hwm_Success )
+        {
+            // If we get an error code that match to the error code defined as require retry,
```

Comment:
This is fine, but I still don't understand WHY we would have to retry a GVD command (specifically, or generally any command).

Reply:
As I explained, this is the flow that was agreed between SW & FW.
Origin from other HW not being ready yet while FW is already ready.
GVD contains other HW information as well, for more info please contact the architect